src/draw.c

```diff
@@ -393,94 +393,58 @@ static int layout_get_height(colored_layout *cl)
         return MAX(h, h_icon);
 }
 
+/**
+ * Create a path on the given cairo context to draw the background of a notification.
+ * The top corners will get rounded by `corner_radius`, if `first` is set.
+ * Respectably the same for `last` with the bottom corners.
+ */
 static void draw_rounded_rect(cairo_t *c, int x, int y, int width, int height, int corner_radius, bool first, bool last)
 {
         const float degrees = M_PI / 180.0;
 
-        if (first && last) {
-                cairo_new_sub_path(c);
-                cairo_arc(c,
-                          x + width - corner_radius,
-                          y + corner_radius,
-                          corner_radius,
-                          -90 * degrees,
-                          0 * degrees);
+        cairo_new_sub_path(c);
+
+        if (last) {
+                // bottom right
                 cairo_arc(c,
                           x + width - corner_radius,
                           y + height - corner_radius,
                           corner_radius,
-                          0 * degrees,
-                          90 * degrees);
+                          degrees * 0,
+                          degrees * 90);
+                // bottom left
                 cairo_arc(c,
                           x + corner_radius,
                           y + height - corner_radius,
                           corner_radius,
-                          90 * degrees,
-                          180 * degrees);
+                          degrees * 90,
+                          degrees * 180);
+        } else {
+                cairo_line_to(c, x + width, y+width);
+                cairo_line_to(c, x, y+width);
```

Comment:
> Let's draw the separator actively. I'm forging the commit right now.

While true it's simple to just draw over it and be done. Things not working like this can indicate some type of problem in the implementation. Like here, you're adding the width to the vertical axis.